From 658edca68df03e7e4ad8abf134672c57215a5615 Mon Sep 17 00:00:00 2001
From: David Scott <dave.scott@eu.citrix.com>
Date: Mon, 8 Sep 2014 15:01:25 +0100
Subject: [PATCH] pygrub: fix non-interactive parsing of grub1 config files

Since c/s d1b93ea2, the type of 'sel' now depends on whether pygrub
is run interactively or not. In non-interactive / quiet mode it is
a string; in interactive mode it is an integer.

When 'sel' is used to index an array it must be used as an integer.

Signed-off-by: David Scott <dave.scott@citrix.com>
diff --git a/tools/pygrub/src/pygrub b/tools/pygrub/src/pygrub
index 1ae34a2..9b9e8de 100644
--- a/tools/pygrub/src/pygrub
+++ b/tools/pygrub/src/pygrub
@@ -639,7 +639,7 @@ def run_grub(file, entry, fs, cfg_args):
         sys.exit(1)
 
     try:
-        img = g.cf.images[sel]
+        img = g.cf.images[int(sel)]
     except IndexError:
         img = g.cf.images[0]
 
